app/code/Bulbulatory/Recomendations/Controller/Adminhtml/Recommendations/Delete.php

```diff
@@ -0,0 +1,35 @@
+<?php
+namespace Bulbulatory\Recomendations\Controller\Adminhtml\Recommendations;
+
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Controller\ResultFactory;
+use Bulbulatory\Recomendations\Api\RecommendationRepositoryInterface;
+
+class Delete extends \Magento\Backend\App\Action
+{
+    private $_recommendationRepositoryInterface;
+
+    public function __construct(
+        Context $context,
+        RecommendationRepositoryInterface $recommendationRepositoryInterface
+    ) {
+        parent::__construct($context);
+        $this->_recommendationRepositoryInterface = $recommendationRepositoryInterface;
+    }
+
+    public function execute() 
+    {
+        try {
+            $id = $this->getRequest()->getParam('recommendation_id');
+            $recommendation = $this->_recommendationRepositoryInterface->getById($id);
+            $this->_recommendationRepositoryInterface->delete($recommendation);
+        } catch (\Exception $e) {
+            $this->messageManager->addErrorMessage('Cannot delete recommendation.');
+        }
```

Comment:
Twoja funkcja `getById($id)` w repository rzuca wyjątek `NoSuchEntityException` gdy nie znajdzie rekomendacji. Można najpierw zrobić catch na niego i wyświetlić inną wiadomość, np. `Recommendation with ID X doesn't exist`, a dopiero potem łapać najogólniejsze `\Exception`.
```suggestion
            $recommendation = $this->_recommendationRepositoryInterface->getById($id);
            $this->_recommendationRepositoryInterface->delete($recommendation);
        } catch (NoSuchEntityException $e) {
            $this->messageManager->addErrorMessage(...);
        } catch (\Exception $e) {
            $this->messageManager->addErrorMessage('Cannot delete recommendation.');
        }
```